x/delayedack/keeper/rollapp_packet.go

```diff
@@ -86,28 +117,64 @@ func (k Keeper) UpdateRollappPacketTransferAddress(
 	// Update rollapp packet with the new updated packet and save in the store
 	rollappPacket.Packet = packet
 	rollappPacket.OriginalTransferTarget = originalTransferTarget
+
+	// Update index: delete the old packet and save the new one
+	err = k.DeletePendingPacketByReceiver(ctx, transferPacketData.Receiver, []byte(rollappPacketKey))
+	if err != nil {
+		return err
+	}
+	err = k.SetPendingPacketByReceiver(ctx, newPacketData.Receiver, rollappPacket.RollappPacketKey())
+	if err != nil {
+		return err
+	}
 	k.SetRollappPacket(ctx, *rollappPacket)
+
 	return nil
 }
 
 // UpdateRollappPacketWithStatus deletes the current rollapp packet and creates a new one with and updated status under a new key.
 // Updating the status should be called only with this method as it effects the key of the packet.
 // The assumption is that the passed rollapp packet status field is not updated directly.
 func (k *Keeper) UpdateRollappPacketWithStatus(ctx sdk.Context, rollappPacket commontypes.RollappPacket, newStatus commontypes.Status) (commontypes.RollappPacket, error) {
-	store := ctx.KVStore(k.storeKey)
+	transferPacketData, err := rollappPacket.GetTransferPacketData()
+	if err != nil {
+		return commontypes.RollappPacket{}, err
+	}
 
-	// Delete the old rollapp packet
 	oldKey := rollappPacket.RollappPacketKey()
+
+	// Update the pending packets index
+	if rollappPacket.Status == commontypes.Status_PENDING {
+		// Delete the old packet from the pending packets by receiver index
+		err = k.DeletePendingPacketByReceiver(ctx, transferPacketData.Receiver, oldKey)
+		if err != nil {
+			return rollappPacket, err
+		}
+	}
+
+	// Delete the old rollapp packet
+	store := ctx.KVStore(k.storeKey)
 	store.Delete(oldKey)
+
 	// Update the packet
 	rollappPacket.Status = newStatus
 	// Create a new rollapp packet with the updated status
 	k.SetRollappPacket(ctx, rollappPacket)
 
-	// Call hook subscribers
 	newKey := rollappPacket.RollappPacketKey()
+
+	// Update the pending packets index
+	if newStatus == commontypes.Status_PENDING {
```

Comment:
can't happen
this method is actually handling PENDING->FINALIZE only

Reply:
fixed! plus added a check at the beginning of the method